### PR TITLE
[feature](merge-cloud) Implement cloud compaction

### PR DESCRIPTION
**Pull request by luwei16**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

File: be/src/cloud/cloud_tablet.h [hunk 1]
```diff
@@ -109,6 +109,69 @@ class CloudTablet final : public BaseTablet {
...
+    void set_last_full_compaction_success_time(int64_t millis) {
+        _last_full_compaction_success_millis = millis;
+    }
+
+    int64_t last_base_compaction_schedule_time() { return _last_base_compaction_schedule_millis; }
+    void set_last_base_compaction_schedule_time(int64_t millis) {
+        _last_base_compaction_schedule_millis = millis;
+    }
+
+    std::vector<RowsetSharedPtr> pick_candidate_rowsets_to_base_compaction();
```
Comment (platoneko) [inline be/src/cloud/cloud_tablet.h:147]: 这个函数应该可以不用了，直接用 `traverse_rowsets`

**Review by platoneko — APPROVED** [review]
Almost LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**